File: apps/SemanticGraph.Visualizer/src/SemanticGraph.Visualizer.Client/Models/GraphEdgeDetailsDto.cs
using System.Text.Json;
using System.Text.Json.Serialization;

namespace SemanticGraph.Visualizer.Client.Models;

public sealed record GraphEdgeDetailsDto(
    [property: JsonPropertyName("edgeId")] string EdgeId,
    [property: JsonPropertyName("relation")] string Relation,
    [property: JsonPropertyName("context")] string? Context,
    [property: JsonPropertyName("confidence")] string Confidence,
    [property: JsonPropertyName("confidenceScore")] double ConfidenceScore,
    [property: JsonPropertyName("weight")] double Weight,
    [property: JsonPropertyName("firstSeenRunId")] long? FirstSeenRunId,
    [property: JsonPropertyName("lastSeenRunId")] long? LastSeenRunId,
    [property: JsonPropertyName("propertiesJson")] JsonElement PropertiesJson,
    [property: JsonPropertyName("source")] GraphEdgeEndpointDto Source,
    [property: JsonPropertyName("target")] GraphEdgeEndpointDto Target,
    [property: JsonPropertyName("evidence")] IReadOnlyList<GraphEdgeEvidenceDto> Evidence);

File: crates/semantic-graph-visualizer-server/src/dto/graph_edge_details_dto.rs
use crate::dto::{GraphEdgeEndpointDto, GraphEdgeEvidenceDto};

use serde::{Deserialize, Serialize};
use serde_json::Value;

#[derive(Debug, Clone, Deserialize, Serialize, PartialEq)]
#[serde(rename_all = "camelCase")]
pub struct GraphEdgeDetailsDto {
    pub edge_id: String,
    pub relation: String,
    pub context: Option<String>,
    pub confidence: String,
    pub confidence_score: f64,
    pub weight: f64,
    pub first_seen_run_id: Option<i64>,
    pub last_seen_run_id: Option<i64>,
    pub properties_json: Value,
    pub source: GraphEdgeEndpointDto,
    pub target: GraphEdgeEndpointDto,
    pub evidence: Vec<GraphEdgeEvidenceDto>,
}
